common/api/core-common.api.md

```diff
@@ -6161,52 +6360,123 @@ export namespace Quantization {
+    // (undocumented)
+    bindPoint2d(indexOrName: string | number, val: Point2d): this;
+    // (undocumented)
+    bindPoint3d(indexOrName: string | number, val: Point3d): this;
+    // (undocumented)
+    bindString(indexOrName: string | number, val: string): this;
+    // (undocumented)
+    bindStruct(indexOrName: string | number, val: object): this;
+    // (undocumented)
+    static from(args: any[] | object | undefined): QueryBinder;
+    // (undocumented)
+    serialize(): object;
+    }
+
+// @beta
 export interface QueryLimit {
-    maxRowAllowed?: number;
-    startRowOffset?: number;
+    count?: number;
+    offset?: number;
 }
 
-// @public
-export enum QueryPriority {
+// @beta
+export interface QueryOptions extends BaseReaderOptions {
+    abbreviateBlobs?: boolean;
+    includeMetaData?: boolean;
+    limit?: QueryLimit;
+    suppressLogErrors?: boolean;
+}
+
+// @beta (undocumented)
+export class QueryOptionsBuilder {
+    // (undocumented)
+    get config(): QueryOptions;
+    // (undocumented)
+    setAbbreviateBlobs(val: boolean): this;
+    // (undocumented)
+    setLimit(val: QueryLimit): this;
+    // (undocumented)
+    setPriority(val: number): this;
+    // (undocumented)
+    setQuota(val: QueryQuota): this;
     // (undocumented)
-    High = 2,
+    setRestartToken(val: string): this;
     // (undocumented)
-    Low = 0,
+    setSuppressLogErrors(val: boolean): this;
     // (undocumented)
-    Normal = 1
+    setUsePrimaryConnection(val: boolean): this;
 }
 
-// @public
+// @beta (undocumented)
+export interface QueryPropertyMetaData {
+    // (undocumented)
+    className: string;
+    // (undocumented)
+    generated: boolean;
+    // (undocumented)
+    index: number;
+    // (undocumented)
+    jsonName: string;
+    // (undocumented)
+    name: string;
+    // (undocumented)
+    system: boolean;
+    // (undocumented)
+    typeName: string;
+}
+
+// @beta
 export interface QueryQuota {
-    maxMemoryAllowed?: number;
-    maxTimeAllowed?: number;
+    memory?: number;
+    time?: number;
 }
 
 // @public
-export interface QueryResponse {
-    // (undocumented)
-    rows: any[];
-    // (undocumented)
-    status: QueryResponseStatus;
+export enum QueryRowFormat {
+    Default = 2,
+    UseArrayIndexes = 2,
+    UseECSqlPropertyNames = 0,
+    UseJsPropertyNames = 1
 }
 
-// @public
-export enum QueryResponseStatus {
+// @beta (undocumented)
+export interface QueryRowProxy {
     // (undocumented)
-    Cancelled = 7,
+    [propertyName: string]: QueryValueType;
     // (undocumented)
-    Done = 2,
+    [propertyIndex: number]: QueryValueType;
     // (undocumented)
-    Error = 5,
+    getMetaData(): QueryPropertyMetaData[];
     // (undocumented)
-    Partial = 3,
+    toArray(): QueryValueType[];
     // (undocumented)
-    PostError = 6,
+    toJsRow(): any;
     // (undocumented)
-    Timeout = 4
+    toRow(): any;
 }
 
+// @beta (undocumented)
+export type QueryValueType = any;

```

Comment:
Should this just be a union of all possible types? Perhaps use a string indexer for structs, with the value being a union again?

Reply:
I am testing this. I ran rush cover and it fail for some other reasons for some ui component  test. I am now doing clean build to see if i can reproduce what you seeing.

==[ @itwin/imodel-components-react ]=============================[ 44 of 79 ]==

D:\bsw\m0\imodeljs\ui\imodel-components\lib\test\setup.js:24
    this.x = x ?? 0;

Reply:
@kabentley talked about that a little while ago...the union would be effectively equivalent to `any` anyway.